src/chainparams.h

```diff
@@ -66,6 +66,126 @@ class CChainParams
     int RejectBlockOutdatedMajority() const { return nRejectBlockOutdatedMajority; }
     int ToCheckBlockUpgradeMajority() const { return nToCheckBlockUpgradeMajority; }
 
+    /**
+     * Scheduled protocol cleanup rule change
+     *
+     * To achieve desired scaling limits, the forward blocks architecture
+     * will eventually trigger a hard-fork modification of the consensuss
+     * rules, for the primary purpoes of dropping enforcement of many
+     * aggregate block limits and altering the difficulty adjustment
+     * algorithm.
+     *
+     * This hard-fork will not activate until it is absolutely necessary
+     * for it to do so, at the point when real demand for additional shard
+     * space in aggregate across all forward block shard-chains exceeds
+     * the available space in the compatibility chain. It is anticipated
+     * that this will not occur until many, many years into the future,
+     * when Freicoin/Tradecraft's usage exceeds even the levels of bitcoin
```

Comment:
In short: exhaustion of the carry-forward balance which feeds the coinbase payout queue. Most of the freicoin presently in existence is in a set-aside fund that was meant to be used for TBD ("to be developed") fairer or more socially responsible issuance mechanisms. I've gotten buy-in from some of the major stakeholders here to "donate" the remaining funds into the carry-forward balance once forward blocks activates. This essentially locks up the funds to be used only to payout cross-shard transfers and things like CT peg-outs, while the actual transfers and peg-outs are locked up in the maturation process. The maturation process includes confirmation and maturation of the shard headers on the compatibility chain, which can't be avoided, but also replay of the transactions onto the compatibility chain. The carry-forward balance is used to pay out in advance of the transaction being replayed on the compatibility chain, but only if funds are available. Once the compatibility blocks are full, the replay queue will get longer and longer, as would the payout delay if there weren't carry-forward funds available.

Since the issuance set-aside is currently roughly twice as many freicoins as are in circulation, putting them into the carry-forward balance will allow fast payouts for a long time into the future, even as the delay in collecting the original funds from the replay of transactions extends into months or possibly even years. Once the fund is exhausted and remains exhausted for some period of time (e.g. a few months), that will be the trigger for activation of the protocol cleanup hard-fork, which among other things finally allows the compatibility chain to grow and/or change difficulty adjustment algorithms to keep up.

Now if you wanted to force activation, you'd have to first grow (via flex cap) at least two forward block shards to have capacity >4MWe/10min, then sustain this long enough to grow the replay backlog by a non-trivial amount, then shuffle funds between the two until there is more funds than the carry-forward balance in flight, and complete this while there is still enough backlog that the carry-forward balance isn't replenished for the months required to achieve activation.

How costly this is to do depends on the exact forward block and activation settings, but when I last did a back of the envelope calculation on some assumed reasonable choices, it would take a large fraction of coins in circulation (e.g. half) and a spam attack lasting 3-5 years to force activation. Or, if there is "real" (aka sustainable) demand the same would be naturally accomplished on the order of a decade or two.

Reply:
I think I see what you're saying, but... why can't a miner just spam transactions to cause this, considering they mine their own fees anyway?

Reply:
They don't collect the fees. The way flex cap works is that they put aside money in order to enlarge the block, which *any* other miner is free to claim by mining a smaller block. So essentially they bank the money as a bounty for smaller blocks, and they never get it back. (Unless they spend 3-5 years 51% attacking the network and not letting other miners mine a single block the entire time. We got bigger problems if that's the case.)

Reply:
Hence the assumption that this translates into "real" demand -- no miner would rationally do what amounts to a free gift to other miners for no purpose if it were a spam attack since the money would just be claimed by some other miner in the very next block, with no net effect. But if there were real demand in terms of fee-paying transactions from users, then they'd rationally give up an equivalent or smaller amount of subsidy to grow the block to collect more fees, and no one would later claim these funds because they themselves would rather collect more fees.

Reply:
Oh, I see what you mean. Yeah, that makes sense.